src/main/utils/upload.ts

```diff
@@ -47,6 +48,96 @@ const getFileStream = async (filePath: string) => {
     return readableStream;
 };
 
+const getZipFileStream = async (
+    zip: StreamZip.StreamZipAsync,
+    filePath: string
+) => {
+    const stream = await zip.stream(filePath);
+    stream.pause();
+
+    let chunkToConsume = new Uint8Array(),
+        chunkArray: number[] = [],
+        closed = false;
+
+    let resolveObj: (value?: any) => void, rejectObj: (reason?: any) => void;
+
+    stream.on('data', (chunk: Uint8Array) => {
+        for (const byte of chunk) {
+            chunkArray.push(byte);
+        }
+        if (chunkArray.length >= FILE_STREAM_CHUNK_SIZE) {
+            chunkToConsume = new Uint8Array(
+                chunkArray.slice(0, FILE_STREAM_CHUNK_SIZE)
+            );
+            chunkArray = chunkArray.slice(FILE_STREAM_CHUNK_SIZE);
+            resolveObj();
+        }
+    });
+
+    stream.on('end', () => {
+        closed = true;
+        resolveObj();
+    });
+
+    const resumeDataStream = () => {
+        return new Promise((resolve, reject) => {
+            resolveObj = resolve;
+            rejectObj = reject;
+            if (closed) {
+                resolveObj();
+            } else {
+                stream.resume();
+            }
+        });
+    };
+
+    const readableStream = new ReadableStream<Uint8Array>({
+        async pull(controller) {
+            resumeDataStream().then(() => {
+                if (chunkToConsume.length > 0) {
+                    controller.enqueue(chunkToConsume);
+                    chunkToConsume = new Uint8Array();
+                } else if (chunkArray.length > 0) {
+                    controller.enqueue(
+                        new Uint8Array(
+                            chunkArray.slice(0, FILE_STREAM_CHUNK_SIZE)
+                        )
+                    );
+                    chunkArray = chunkArray.slice(FILE_STREAM_CHUNK_SIZE);
+                } else {
+                    controller.close();
+                }
+                stream.pause();
```

Comment:
what happens if while we are processing this chunk, a second data event comes?

Reply:
Right, so should I move `stream.pause()` to the top of this function?

Reply:
ohh okk.. got it , just ensure that we read any data left in the chunkArray after the stream ends

Reply:
don't know this event-based approach doesn't seem right here, if we got something like the [reader](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/getReader)  we have in the browser, 
reading this doc [https://nodejs.dev/learn/nodejs-streams] it seems there is a way , could you just investigate this a bit and check if we could writable stream as they mention there to read the stream

Reply:
I tried to find something like `read` and also with the other forms of streams, but I wasn't able to make it work. 😥